src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -331,11 +331,17 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
         onSubmitSuccess(resp, model);
       }
     } catch (err) {
-      const apiErrors = Array.isArray(err?.responseJSON)
-        ? `: ${err.responseJSON.join(', ')}`
-        : err?.responseJSON?.nonFieldErrors
-        ? `: ${err.responseJSON.nonFieldErrors.join(', ')}`
-        : '';
+      let errors: string[] = [];
+      if (Array.isArray(err?.responseJSON)) {
+        errors = err.responseJSON;
+      } else if (err?.responseJSON && Object.entries(err?.responseJSON)) {
+        for (const [, value] of Object.entries(err?.responseJSON)) {
+          if (typeof value === 'string') {
+            errors.push(value as string);
+          }
+        }
```

Comment:
We prefer using [Array.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)/forEach when iterating over lists 

You can also use `Object.values()` since you're not using the keys.

In this case, I think you can actually use [Array.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)